Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Killing Plot Adaptation

Fixes #1141, closes #499.

## Changelog:
- **Restricted some normal killing plots:**
  - Only those who can drink wine and organize banquets (so not tribal, nomadic rulers, demons, undead, etc.), can be killed by the **Poisoned Wine** plot.
  - Only humanoids who can't fly and like to watch the sunrise (so not harpies, Necromantic (religion) characters, etc.) can be killed by the **Falling Accident** plot.
  - Only those who like to show in public and sit in a carriage (so not nomadic, tribal, Necromantic (religion) characters, centaurs, etc.) can be killed by the **Archer Ambush** plot.
  - Only those who like to show in public and enter forgotten taverns (so not nomadic, tribal, Necromantic (religion) characters, etc.) can be killed by the **Manure Explosion** plot.
  - Only humanoids who can't breathe underwater (so not murlocs, undead, etc.) can be killed by the **Bathtub Drowning** plot.
  - Only those who can't breathe underwater or fly (so not murlocs, undead, harpies, etc.) can be killed by the **Boating Accident** plot.
  - Only those who aren't immune to natural venom (so not demons, undead, etc.) be killed by the **Poisonous Snake** plot.
  - Only those who can't fly and like to watch the sunrise (so not harpies, Necromantic (religion), tribal, nomadic characters, etc.) can be killed by the **Balcony Plummet** plot.
  - Only those who can hunt (so not **Elunic** religion, etc.) can be killed by the **Hunting Accident** plot.
  - Only humanoids who can't fly and sit in a carriage (so not harpies, centaurs, etc.) can be killed by the **Сarriage Accident** plot.
- **Restricted some killing in hiding plots:**
  - Only those who can get sick (so not undead, demons, etc.), can't be killed by the **Poisoned Food** plot.
  - Only those who aren't immune to natural venom (so not demons, undead, etc.) be killed by the **Snake Charmer** plot.
  - Only humanoids can be killed by the **Bribed Nurse** plot.
- Reworked the **Disguised Highwaymen** plot. It can fire for children and incapable characters. Plus, instead of highwaymen, it may mention:
  - Dragon hunters if the plot target is a dragon (for the future, you can't plot against dragons right now).
  - Monster hunters if the plot target is a demon, undead, void being, worgen.
  - Witch hunters if the plot target practices Dark magic.
- Reworked the **Ninja** plot. It can fire for incapable characters.


# How to test:
Play and check how plots work for different characters.

I added a game rule to **SLIGHTLY** increase the chances of killing plot attempts (of player only or both player and AI) to save your time. The rule is called **Killing Chances** and can be found at the end of the game rules.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/75096083-4c8d6d80-55b5-11ea-96e7-18b490120c6e.png)

</details>

Note: the game rule is here to save your time, it's not an object of testing.

Also, I recommend to write `yesman` in the console, so you can invite everyone in your plots.
